yarn-project/aztec-node/src/aztec-node/server.ts

```diff
@@ -903,10 +903,8 @@ export class AztecNodeService implements AztecNode, Traceable {
     return this.contractDataSource.addContractClass(contractClass);
   }
 
-  public addContractArtifact(address: AztecAddress, artifact: ContractArtifact): Promise<void> {
-    this.log.info(`Adding contract artifact ${artifact.name} for ${address.toString()} via API`);
-    // TODO: Node should validate the artifact before accepting it
-    return this.contractDataSource.addContractArtifact(address, artifact);
+  public registerContractFunctionNames(_address: AztecAddress, names: Record<string, string>): Promise<void> {
+    return this.contractDataSource.registerContractFunctionNames(names);
```

Comment:
Instead of having the caller pass the selector and name, I think it'd be best if they pass the full function signature, so the server computes the selector from it and extracts the name. This way we don't risk a malicious user overwriting valid selectors with trash.

And I'm fine pushing this for later, so feel free to create an issue instead of addressing now.